pkgs/servers/jellyfin/update.sh

```diff
@@ -15,23 +15,36 @@ if [[ "$currentVersion" == "$latestVersion" ]]; then
 fi
 
 pushd "$(dirname "${BASH_SOURCE[0]}")"
-nugetDepsFile=$(realpath ./nuget-deps.nix)
+realDir=$(realpath ./)
 popd
 
 update-source-version jellyfin "$latestVersion"
 
 store_src="$(nix-build . -A jellyfin.src --no-out-link)"
-src="$(mktemp -d /tmp/jellyfin-src.XXX)"
-echo "Temp src dir: $src"
-cp -rT "$store_src" "$src"
-chmod -R +w "$src"
 
-pushd "$src"
+update_nuget_for_arch () {
+    src="$(mktemp -d /tmp/jellyfin-src.XXX)"
+    echo "Temp src dir: $src"
+    cp -rT "$store_src" "$src"
+    chmod -R +w "$src"
 
-mkdir ./nuget_tmp.packages
-dotnet restore Jellyfin.Server --packages ./nuget_tmp.packages --runtime linux-x64
 
-nuget-to-nix ./nuget_tmp.packages > "$nugetDepsFile"
+    pushd "$src"
 
-popd
-rm -r "$src"
+    mkdir ./nuget_tmp.packages
+
+    nugetDepsFile="$realDir/nuget-deps-"$1".nix"
+
+    echo "Updating depencies for $1 at $nugetDepsFile"
+
+    dotnet restore Jellyfin.Server --packages ./nuget_tmp.packages --runtime linux-"$1"
+    nuget-to-nix ./nuget_tmp.packages > "$nugetDepsFile"
```

Comment:
If you move the `nuget-to-nix` invocation *after* all the various `update_nuget_for_arch` invocations, you can streamline the dependencies by generating only one `deps.nix` file. This file will have all the packages restore for the various RIDs, and will be equivalent to calling `dotnet restore` with multiple RIDS, as referenced by https://github.com/NixOS/nixpkgs/pull/177902#pullrequestreview-1011567105

Reply:
Thanks for the suggestion, it greatly simplifies the changes